services/src/datasets/external/nfdi/mod.rs

```diff
@@ -0,0 +1,1031 @@
+    fn create_raster_result_descriptor(
+        crs: SpatialReferenceOption,
+        info: &RasterInfo,
+    ) -> RasterResultDescriptor {
+        RasterResultDescriptor {
+            data_type: info.data_type,
+            spatial_reference: crs,
+            measurement: info
+                .measurement
+                .as_ref()
+                .map_or(Measurement::Unitless, Clone::clone),
+            no_data_value: info.no_data_value,
+        }
+    }
+
+    /// Retrieves a file-object from the core-storage. It assumes, that the dataset consists
+    /// only of a single object group with a single object (the file).
+    async fn get_single_file_object(&self, id: &DatasetId) -> Result<Object> {
+        let mut ds_stub = self.dataset_stub.clone();
+
+        let group = ds_stub
+            .get_dataset_object_groups(GetDatasetObjectGroupsRequest {
+                id: Self::dataset_nfdi_id(id)?,
+            })
+            .await?
+            .into_inner()
+            .object_groups
+            .into_iter()
+            .next()
+            .ok_or(Error::NoMainFileCandidateFound)?;
+
+        group
+            .objects
+            .into_iter()
+            .next()
+            .ok_or(Error::NoMainFileCandidateFound)
+    }
+
+    /// Creates the loading template for vector files. This is basically a loading
+    /// info with a placeholder for the download-url. It will be replaced with
+    /// a concrete url on every call to `MetaData.loading_info()`.
+    /// This is required, since download links from the core-storage are only valid
+    /// for 15 minutes.
+    fn vector_loading_template(
+        layer_name: String,
+        rd: &VectorResultDescriptor,
+    ) -> OgrSourceDataset {
+        let data_type = match rd.data_type {
+            VectorDataType::Data => None,
+            x => Some(x),
+        };
+
+        // Map column definition
+        let mut int = vec![];
+        let mut float = vec![];
+        let mut text = vec![];
+
+        for (k, v) in &rd.columns {
+            match v {
+                FeatureDataType::Category | FeatureDataType::Int => int.push(k.to_string()),
+                FeatureDataType::Float => float.push(k.to_string()),
+                FeatureDataType::Text => text.push(k.to_string()),
+            }
+        }
+
+        let link = format!("/vsicurl/{}", URL_REPLACEMENT);
+
+        let column_spec = OgrSourceColumnSpec {
+            format_specifics: None,
+            x: "".to_string(),
+            y: None,
+            int,
+            float,
+            text,
+            rename: None,
+        };
+
+        let time = OgrSourceDatasetTimeType::None;
+
+        OgrSourceDataset {
+            file_name: PathBuf::from(link),
+            layer_name,
+            data_type,
+            time,
+            default_geometry: None,
+            columns: Some(column_spec),
+            force_ogr_time_filter: false,
+            force_ogr_spatial_filter: false,
```

Comment:
Do filters work or do we have to force this?

Reply:
Yes, I just tested it.